packages/schematics/angular/migrations/update-10/update-tslint.ts

```diff
@@ -0,0 +1,147 @@
+import { readJsonFileAsAstObject } from '../update-9/utils';
+
+export const TSLINT_VERSION = '~6.1.0';
+const TSLINT_CONFIG_PATH = '/tslint.json';
+
+const RULES_TO_DELETE: string[] = [
+  'no-use-before-declare',
+  'no-unused-variable',
+];
+
+const RULES_TO_ADD: Record<string, JsonValue> = {
+  align: {
+    options: ['parameters', 'statements'],
+  },
+  'arrow-return-shorthand': true,
+  curly: true,
+  eofline: true,
+  'import-spacing': true,
+  indent: {
+    options: ['spaces'],
+  },
+  semicolon: { options: ['always'] },
+  'space-before-function-paren': {
+    options: {
+      anonymous: 'never',
+      asyncArrow: 'always',
+      constructor: 'never',
+      method: 'never',
+      named: 'never',
+    },
+  },
+  'typedef-whitespace': {
+    options: [
+      {
+        'call-signature': 'nospace',
+        'index-signature': 'nospace',
+        parameter: 'nospace',
+        'property-declaration': 'nospace',
+        'variable-declaration': 'nospace',
+      },
+      {
+        'call-signature': 'onespace',
+        'index-signature': 'onespace',
+        parameter: 'onespace',
+        'property-declaration': 'onespace',
+        'variable-declaration': 'onespace',
+      },
+    ],
+  },
+  whitespace: {
+    options: [
+      'check-branch',
+      'check-decl',
+      'check-operator',
+      'check-separator',
+      'check-type',
+      'check-typecast',
+    ],
+  },
+};
+
+export default function (): Rule {
+  return (tree, context) => {
+    const logger = context.logger;
+
+    // Update tslint dependency
+    const current = getPackageJsonDependency(tree, 'tslint');
+
+    if (!current) {
+      logger.warn('"tslint" in not a dependency of this workspace.');
+
+      return;
+    }
+
+    if (current.version !== TSLINT_VERSION) {
+      addPackageJsonDependency(tree, {
+        type: current.type,
+        name: 'tslint',
+        version: TSLINT_VERSION,
+        overwrite: true,
+      });
+    }
+
+    // Update tslint config.
+    const tslintJsonAst = readJsonFileAsAstObject(tree, TSLINT_CONFIG_PATH);
+    if (!tslintJsonAst) {
+      logger.warn('Cannot find "tslint.json".');
+
+      return;
+    }
+
+    // Remove old/deprecated rules.
+    for (const rule of RULES_TO_DELETE) {
+      const tslintJsonAst = readJsonFileAsAstObject(tree, TSLINT_CONFIG_PATH) as JsonAstObject;
+      const rulesAst = findPropertyInAstObject(tslintJsonAst, 'rules');
+      if (rulesAst?.kind !== 'object') {
+        break;
+      }
+
+      const recorder = tree.beginUpdate(TSLINT_CONFIG_PATH);
+      removePropertyInAstObject(recorder, rulesAst, rule);
+      tree.commitUpdate(recorder);
+    }
+
+    // Add new rules only if the configuration doesn't extend 'tslint-config-prettier'.
+    // This is because some rules conflict with prettier.
+    const extendsAst = findPropertyInAstObject(tslintJsonAst, 'extends');
```

Comment:
If an extends property within the extension hierarchy does not contain `tslint:recommended` then this part should be skipped.  Since this is attempting to augment upstream changes to the `tslint:recommended` ruleset to ensure a stable ruleset after update, without its presence there should be nothing to augment.